src/filters/jekyll/array.rs

```diff
@@ -0,0 +1,316 @@
+use filters::invalid_input;
+use liquid_compiler::{Filter, FilterParameters};
+use liquid_derive::*;
+use liquid_error::Result;
+use liquid_interpreter::Context;
+use liquid_interpreter::Expression;
+use liquid_value::Value;
+use std::fmt::Write;
+
+#[derive(Debug, FilterParameters)]
+struct PushArgs {
+    #[parameter(description = "The element to append to the array.")]
+    element: Expression,
+}
+
+#[derive(Clone, ParseFilter, FilterReflection)]
+#[filter(
+    name = "push",
+    description = "Appends the given element to the end of an array.",
+    parameters(PushArgs),
+    parsed(PushFilter)
+)]
+pub struct Push;
+
+#[derive(Debug, FromFilterParameters, Display_filter)]
+#[name = "push"]
+struct PushFilter {
+    #[parameters]
+    args: PushArgs,
+}
+
+impl Filter for PushFilter {
+    fn evaluate(&self, input: &Value, context: &Context) -> Result<Value> {
+        let args = self.args.evaluate(context)?;
+
+        let element = args.element.clone();
```

Comment:
`args.element` is a reference which is why we need the `clone`, right?

Reply:
Yes. Removing that `clone` gives:
```
error[E0308]: mismatched types
  --> src\filters\jekyll\array.rs:41:20
   |
41 |         array.push(element);
   |                    ^^^^^^^ expected enum `liquid_value::Value`, found reference
   |
   = note: expected type `liquid_value::Value`
              found type `&liquid_value::Value`
```